staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -141,12 +142,28 @@ func (c *healthzCheck) Check(r *http.Request) error {
 	return c.check(r)
 }
 
+// getExcludedChecks extracts the health check names to be excluded from the query param
+func getExcludedChecks(r *http.Request) sets.String {
+	checks, found := r.URL.Query()["exclude"]
```

Comment:
> An exclude system means that if a provider wants a static list of checks they need to revisit the exclusion list with every release.

For something like permissions, I agree with you. For something like health checks, I think the default should be to pick up new health checks added to the apiserver. A provider specifying an inclusion list would have to revisit it every release as well to ensure they weren't missing new named health checks added to the server that were critical.

> The difficulty would be if you want remove an item but automatically pick up new checks without looking. This seems like less of a problem.

That actually sounds like exactly the use case https://github.com/kubernetes/kubernetes/issues/70591 describes ... opting out of a specific health check because of higher-level knowledge of the deployment, but wanting all the other built-in health checks to run.

Reply:
With whitelisting, don't we encounter the same issues that @deads2k mentioned earlier in the thread, re: version skew and rollbacks?